services/api-server/src/simcore_service_api_server/modules/webserver.py

```diff
@@ -184,12 +145,60 @@ async def list_projects(self, solver_name: str) -> List[Project]:
         return list(projects)
 
 
-# TODO: init client and then build sessions from client using depenencies
-#
-# from ..utils.client_base import BaseServiceClientApi
-# class WebserverApi(BaseServiceClientApi):
-#     """ One instance per app """
+def _get_secret_key(settings: WebServerSettings):
+    secret_key_bytes = settings.session_secret_key.get_secret_value().encode("utf-8")
+    while len(secret_key_bytes) < 32:
+        secret_key_bytes += secret_key_bytes
+    secret_key = secret_key_bytes[:32]
+
+    if isinstance(secret_key, str):
+        pass
+    elif isinstance(secret_key, (bytes, bytearray)):
+        secret_key = base64.urlsafe_b64encode(secret_key)
+    return secret_key
+
+
+class WebserverApi(BaseServiceClientApi):
+    """Access to web-server API"""
+
+    # def create_auth_session(self, session_cookies) -> AuthSession:
+    #     """ Needed per request, so it can perform """
+    #     return AuthSession(client=self.client, vtag="v0", session_cookies=session_cookies)
+
+
+# MODULES APP SETUP -------------------------------------------------------------
+
+
+def setup(app: FastAPI, settings: Optional[WebServerSettings] = None) -> None:
+    if not settings:
+        settings = WebServerSettings()
+
+    assert settings is not None  # nosec
 
-#     def create_auth_session(self, session_cookies) -> AuthSession:
-#         """ Needed per request, so it can perform """
-#         return AuthSession(client=self.client, vtag="v0", session_cookies=session_cookies)
+    setup_client_instance(
+        app, WebserverApi, api_baseurl=settings.base_url, service_name="webserver"
+    )
+
+    # TODO: old startup. need to integrat
+    # TODO: init client and then build sessions from client using depenencies
```

Comment:
do you need to take care of this before finishing this PR?

Reply:
I would like to leave the as reminders for next iteration (see #2344 ) . Otherwise this PR is cluttered with too many features and is difficult to review/understand